### DISCUSSION TITLE
typing a ♦

### DISCUSSION THREAD

**Discussion by wordmuse** [freeplane/freeplane#273] [Ideas]
Pressing Alt+4 (on the number keypad) lets me type a diamond (♦). When I do it in a browser, I get the diamond. If I copy the diamond from here, I can paste it inline, inside a Freeplane node.  But if I press Alt+4 while in the node, I get this: _?_ - and not ♦.

What's going on? Is there some way that I can type a diamond character while in Freeplane, or do I have to first type it somewhere else and then paste it into FP?

The use case is that I want to mark various sentences within nodes as of "diamond importance." If it can't currently be done, could it be considered as a feature request?

Hmmm - when I type Alt+4, I get a black diamond. When I save this post, the diamond turns red. I'm not sure what's going on with that.

~ Bal

**Comment by rexelbartolome** [comment]
I'm not sure I understand it myself exactly but basically an emoji/dingbat like ♦ can either be interpreted by the font you're typing in, by the app/program, or by the OS you're on. The reason why the diamond is black when typing in github is the same reason why it's black when it's typed on a Notepad, it's being rendered as a default Unicode character (usually just black and white) even if it's not supported in the font you're typing in. If it has color, then it's likely either the app you're on or the OS you're on interpreted that Unicode correctly as a colored emoji (you also get a different style when you use the same emoji in Twitter for example, since they have their own way of rendering these characters)

If it's a question mark then my guess would be it's trying to find that Unicode character in the font you're using but your font doesn't support those types of characters, so it doesn't know how to render it.

**Comment by quickfold** [comment]
What is your OS? This is probably something that can be most easily accomplished with another program, e.g. if you are on Windows, you could use AutoHotKey.

**Comment by wordmuse** [comment]
Sorry for the lateness of the response... just saw your comments now.  

**rexelbartolome -** Yes - I get question marks, so what you said makes sense.

**quickfold** - My OS is Windows 10. I have AutoHotkey. And I'm really new with it, so I will look to see if I can figure it out.

Thanks.

~ Bal

**Reply by quickfold** [reply]
If you can type the diamond in Notepad, you can create an AHK script that is just:

`::.dia::<>`

Where the `.dia` can be replaced with any string you like, and `<>` is the actual diamond you want. When running, this script should automatically replace `.dia` with the diamond in any program after you hit the spacebar.

**Reply by rexelbartolome** [reply]
Hmmm that didn't work in my case

https://user-images.githubusercontent.com/88552647/161904509-d46a6089-1e55-44ac-9f51-a290e73ef96d.mp4

I think it needs something else for this to work inside freeplane nodes. Maybe a way for freeplane to read the emojis properly? https://bobsguides.com/convert-db-utf8.html Don't really understand it all that much, maybe @dpolivaev could explain whats happening

**Reply by quickfold** [reply]
Sorry, I didn't mean for you to include the `<>`—that was just my way of "drawing" an ASCII diamond! It's more like ::.dia::@ where `@` is replaced by the diamond glyph.

**Comment by wordmuse** [comment]
I can type the ♦ into Notepad using Alt+4 (on the number pad), which is what I did just now. And I can then copy and paste the diamond into a node in Freeplane.  What I can't do is simply use Alt+4 to type directly into the Freeplane node. When I copy and paste, am I pasting a graphic rather than a character?

~ Bal

**Comment by EdoFro** [comment]
I have no problem using alt + 4 to get a ♦ in freeplane, in node's text, details or note. In the filterbar, in the text box I can also insert the ♦.

I think it may be an issue with the font or the node style (or maybe the template)  you are using.

Can you attach a mindmap with the problem?

Maybe its not a Freeplane issue, but an issue with your java version.

**Reply by rexelbartolome** [reply]
I see, it may be a font issue then? The font **SansSerif** seems to be compatible with lots of Unicode emoticons/dingbats and can therefore render it properly. Other fonts rely on a fallback that the program could interpret (i.e., Notion interprets emojis properly regardless of the font).

https://user-images.githubusercontent.com/88552647/162114917-a55cd9f6-4da9-403a-84df-bad2ac22d4d4.mp4

**Comment by macmarrum** [comment]
@wordmuse This is a cool trick, entering `♦` with Alt+4 (on Windows). My private computer has Linux on board, therefore I wanted to research the topic to see what is available there. Wikipedia has an interesting article about this subject https://en.wikipedia.org/wiki/Alt_code.
The way I understand the matter, it's up to an application to interpret the Alt+code sequence.

Interestingly, on my work-related Windows 10, in Freeplane 1.10.1 pre04 using Java 17, Alt+4 produces `¦` (U+00A6). On the same system, in jEdit using the same Java 17, Alt+4 produces `♦` (U+2666).
